msal/application.py

```diff
@@ -830,16 +885,17 @@ def acquire_token_by_username_password(
             CLIENT_CURRENT_TELEMETRY: _build_current_telemetry_request_header(
                 self.ACQUIRE_TOKEN_BY_USERNAME_PASSWORD_ID),
             }
+        claims = _merge_claims_and_capabilities(self.client_capabilities, claims)
         if not self.authority.is_adfs:
             user_realm_result = self.authority.user_realm_discovery(
                 username, correlation_id=headers[CLIENT_REQUEST_ID])
             if user_realm_result.get("account_type") == "Federated":
                 return self._acquire_token_by_username_password_federated(
                     user_realm_result, username, password, scopes=scopes,
-                    headers=headers, **kwargs)
+                    headers=headers, data=dict(kwargs.pop("data", {}), claims=claims), **kwargs)
```

Comment:
The new addition is already significantly long enough to occupy a line. We do not squeeze it into exist line, then. (Same suggestion on elsewhere, too.)
```suggestion
                    data=dict(kwargs.pop("data", {}), claims=claims),
                    headers=headers, **kwargs)
```